components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.iterate.ui/src/main/resources/web/iterate-mediator/update-mediator.jsp

```diff
@@ -31,10 +37,33 @@
     String continueParent =  request.getParameter("continueParent");
     String preservePayload = request.getParameter("preservePayload");
     String sequentialMed = request.getParameter("sequentialMed");
-    
+
     XPathFactory xPathFactory = XPathFactory.getInstance();
-    iterateMediator.setExpression(xPathFactory.createSynapseXPath("itr_expression", request, session));
-    iterateMediator.setAttachPath(xPathFactory.createSynapseXPath("attach_path", request, session));
+    SynapseJsonPathFactory jsonPathFactory = SynapseJsonPathFactory.getInstance();
+
+    String iterateExpression = request.getParameter(ITR_EXPRESSION_KEY);
+    if (iterateExpression.startsWith(JSON_EVAL_START_STRING)) {
+        iterateMediator.setExpression(
+                jsonPathFactory.createSynapseJsonPath(ITR_EXPRESSION_KEY,
+                        iterateExpression.trim().substring(
+                                JSON_EVAL_START_STRING.length(), iterateExpression.length() - 1)));
+    } else {
+        iterateMediator.setExpression(xPathFactory.createSynapseXPath(ITR_EXPRESSION_KEY, iterateExpression, session));
+    }
+
+    String attachPathExpression = request.getParameter(ATTACH_PTH_KEY);
+    if (attachPathExpression == null || "".equals(attachPathExpression)) {
```

Comment:
shall we use StringUtils.isEmpty() for this check?

Reply:
+1. 
Fixed